src/input.cpp

```diff
@@ -734,9 +734,22 @@ std::string input_context::get_available_single_char_hotkeys( std::string reques
     return requested_keys;
 }
 
+const input_context::input_event_filter input_context::disallow_lower_case =
+[]( const input_event &evt ) -> bool {
```

Comment:
These don't have to be lambdas.  They could be regular functions (or static member functions of `input_context`) and they could still be assigned to a `std::function`.

Reply:
Yeah Indeed. I was thinking about moving the lambda from parameter lists to the class scope but didn't realize it could be made a static function instead.